front/lib/api/llm/clients/fireworks/index.ts

```diff
@@ -0,0 +1,82 @@
+import OpenAI from "openai";
+
+import type { AgentActionSpecification } from "@app/lib/actions/types/agent";
+import type { FireworksWhitelistedModelId } from "@app/lib/api/llm/clients/fireworks/types";
+import {
+  toMessage,
+  toTool,
+} from "@app/lib/api/llm/clients/fireworks/utils/conversation_to_openai";
+import { streamLLMEvents } from "@app/lib/api/llm/clients/fireworks/utils/openai_to_events";
+import { LLM } from "@app/lib/api/llm/llm";
+import type { LLMEvent } from "@app/lib/api/llm/types/events";
+import type {
+  LLMClientMetadata,
+  LLMParameters,
+} from "@app/lib/api/llm/types/options";
+import type { ModelConversationTypeMultiActions } from "@app/types";
+import { dustManagedCredentials } from "@app/types";
+
+export class FireworksLLM extends LLM {
+  private client: OpenAI;
+
+  private metadata: LLMClientMetadata = {
+    clientId: "fireworks",
```

Comment:
`openai_chat_completion`

Not likely but we might add other ai providers (for example deepseek relies on chat completion as well)